src/poll/poll.controller.ts

```diff
@@ -59,8 +65,12 @@ export class PollController {
                 }
             }
         }
+
         const poll = await this.mongoService.createPoll(params);
-        await this.sseService.emit({
+
+        if (Number(process.env.CACHE)) this.voteRequestHandlerService.cacheVotePowers(poll);
```

Comment:
So this will precache all voters power? Have you seen any rate limit issues or anything?

I would like to see enhancements like this as a separate PR in the future

Reply:
Yes and it will run async so the poll is available even if the caching hasn't finished yet. I wasn't quite sure yet what would be the best approach to narrow down the user count by community since it would require getting a set of users per discord server. I found two problems with that: 1 As of now the backend does not use discordjs and can't query users per guild directly. We'll have to decide if we want to add discordjs or query it through the bot or something else 2 If we implement voting via FE we might have different filter other than discord guild (i.e. by governator community)

Reply:
Maybe for now we cache vote power on initial vote instead of during poll creation. Thoughts?

Reply:
Can you elaborate a bit more? For what I understsnd it would still be the same set of users? I think there could be situations where this could lead to a load spike for example when a poll gets first announced and there would be some hype.

Reply:
That's a good point. This solution handles load spiking well. 

Does it refresh cache if user adds/removes eth accounts?

When new users create account, are they able to vote?

Reply:
It does not refresh, will add that. New users can vote but uncached.

Reply:
Actually no, this is fine. New accounts will not be cached but count accurately. Removed accounts will remain in cache until the poll ends, but not be counted. I will not implement updating for now because caching might not be needed anymore if we go for graph based implementation of strategies.

Reply:
On a second thought, it would lead to most new users not being cached for their first poll which is pretty bad UX.. probably worth while doing then.

Reply:
Let's pull caching out of this PR and into a new branch so we can get the other units in.

Reply:
Was easier to fix this than to split the PR